jerry-core/ecma/base/ecma-gc.c

```diff
@@ -315,7 +315,8 @@ ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
         case ECMA_INTERNAL_PROPERTY_SCOPE: /* a lexical environment */
         case ECMA_INTERNAL_PROPERTY_PARAMETERS_MAP: /* an object */
         {
-          ecma_object_t *obj_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, property_value);
+
+          ecma_object_t *obj_p = ecma_get_object_from_value (property_value);
```

Comment:
Unfortunately this is still a slow-down, since you also add object flags to the value. I would still prefer a variant to the ECMA_GET_NON_NULL_POINTER macro.

ECMA_INTERNAL_VALUE_GET_POINTER / ECMA_INTERNAL_VALUE_SET_POINTER